pilot/pkg/xds/ecds.go

```diff
@@ -167,13 +167,11 @@ func referencedSecrets(proxy *model.Proxy, push *model.PushContext, resourceName
 	// TODO: we get the WasmPlugins here to get the secrets reference in order to decide whether ECDS push is needed,
 	//       and we will get it again at extension config build. Avoid getting it twice if this becomes a problem.
 	watched := sets.New(resourceNames...)
-	wasmPlugins := push.WasmPlugins(proxy)
+	wasmPlugins := push.WasmPluginsByName(proxy, core.ParseExtensionName(resourceNames))
```

Comment:
I am not sure this is secure?

One is saying: "the proxy can access and pull secrets referenced by WasmPlugins that apply to it"

The new code is saying: "the proxy can access any pull secrets referenced by _any_ WasmPlugin they request, even if the WasmPlugin doesn't apply to them". i.e. in a different namespace, etc?

Reply:
good catch, this can cause secret escalate

Reply:
I think this depends on "Does istiod trust the connected xDS client?"
If we don't trust xDS client at all, it can use fake labels(bypass WorkloadSelector check) to get a WasmPlugin in http listener. And this can also cause secret escalate.
I'm not sure if such strict validation is necessary here. I'm happy to hear your opinions.

Reply:
I trust you bu also limit what you can do, think about this something like rbac

Reply:
I have two ideas:

One is to enhance push.WasmPlugins(proxy) to allow it to obtain the service information of the waypoint proxy, and further filter the corresponding WasmPlugin based on these services.

The other is to directly use the SubjectAccessReview to get RBAC permissions for the obtained secret, similar to how secrets are mounted in gateways. This approach might change the current behavior but would be more secure.

I would like to hear your thoughts on these two ideas！😊

Reply:
`WasmPluginsByListenerInfo` maybe used to get the wasmplugins

Reply:
* Refactor WorkloadPolicyMatcher, make it support multi-service. 
* In WasmPlugins(proxy *Proxy), if current proxy is a waypoint, query service in ambientIndex.